packages/visual-dev/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import styled from 'styled-components'
+
+import * as Styles from './Styles'
+
+interface SwitchProps {
+	id: string
+	status?: "on" | "off"
+	color?: "success" | "critical"
+}
+
+const SwitchDiv = styled.div`
+	${Styles.wrapper}
+`;
+
+const SwitchLabel = styled.label`
+	${Styles.base}
+`;
+
+const SwitchBox = styled.input<Required<{color:string}>>`
+	${Styles.off}
+	&:checked + ${SwitchLabel} {
+		${props => Styles[props.color]}
+		${Styles.on}
+	}
+`;
+
+export const Switch: React.FC<SwitchProps> = ({
```

Comment:
With these types of components it's important for them to be controlled from the outside (i.e. it's parent component or something similar). This would mean adding a `checked` and `onChange` prop so that a parent component can control when it is toggled or not (particularly needed when used in forms) and be able to know when it's been toggled.

Check out this example for how this component should be used: https://headlessui.dev/react/switch